### PR TITLE
Fixes: 8588

### PR DESCRIPTION
**Pull request by john-bodley**
### Description

Addresses https://github.com/sqlalchemy/sqlalchemy/issues/8588.

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [x] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Comment by zzzeek** [comment]
thanks for this.   are you interested in fixing this so that it can be tested?  right now it can't be tested because the regex is hardcoded inside of base.py when it should be one of the regexes in mysql/reflection.py.      Once moved there,  we have a suite of tests against the regex units directly [here](https://github.com/sqlalchemy/sqlalchemy/blob/eddf474d528f55a2ed56e3dac1b0e5decd1e0952/test/dialect/mysql/test_reflection.py#L1287).  

Here's a patch to start:

```diff
diff --git a/lib/sqlalchemy/dialects/mysql/base.py b/lib/sqlalchemy/dialects/mysql/base.py
index a3e99514bc..e8ddb6d1e9 100644
--- a/lib/sqlalchemy/dialects/mysql/base.py
+++ b/lib/sqlalchemy/dialects/mysql/base.py
@@ -3162,7 +3162,7 @@ class MySQLDialect(default.DefaultDialect):
         sql = self._show_create_table(
             connection, None, charset, full_name=full_name
         )
-        if re.match(r"^CREATE (?:ALGORITHM)?.* VIEW", sql):
+        if parser._check_view(sql):
             # Adapt views to something table-like.
             columns = self._describe_table(
                 connection, None, charset, full_name=full_name
diff --git a/lib/sqlalchemy/dialects/mysql/reflection.py b/lib/sqlalchemy/dialects/mysql/reflection.py
index e7a6b157fc..8e9f7884be 100644
--- a/lib/sqlalchemy/dialects/mysql/reflection.py
+++ b/lib/sqlalchemy/dialects/mysql/reflection.py
@@ -74,6 +74,9 @@ class MySQLTableDefinitionParser:
                     pass
         return state
 
+    def _check_view(self, sql):
+        return bool(self._re_is_view.match(sql))
+
     def _parse_constraints(self, line):
         """Parse a KEY or CONSTRAINT line.
 
@@ -409,6 +412,9 @@ class MySQLTableDefinitionParser:
             self.preparer._unescape_identifier,
         )
 
+        self._re_is_view = _re_compile(
+            r"^CREATE (?:ALGORITHM)?.* VIEW"
+        )
         # `col`,`col2`(32),`col3`(15) DESC
         #
         self._re_keyexprs = _re_compile(
diff --git a/test/dialect/mysql/test_reflection.py b/test/dialect/mysql/test_reflection.py
index f815a7b3c9..5f3db615ba 100644
--- a/test/dialect/mysql/test_reflection.py
+++ b/test/dialect/mysql/test_reflection.py
@@ -1285,8 +1285,6 @@ class ReflectionTest(fixtures.TestBase, AssertsCompiledSQL):
 
 
 class RawReflectionTest(fixtures.TestBase):
-    __backend__ = True
-
     def setup_test(self):
         dialect = mysql.dialect()
         self.parser = _reflection.MySQLTableDefinitionParser(
@@ -1412,3 +1410,14 @@ class RawReflectionTest(fixtures.TestBase):
                 "SET NULL",
             ),
         )
+
+    @testing.combinations(
+        (
+            "CREATE ALGORITHM=UNDEFINED DEFINER=`scott`@`%` SQL SECURITY DEFINER VIEW `v4` AS select",
+            True,
+        ),
+        ("CREATE VIEW `v4` AS select", True),
+        ("CREATE TABLE `v4`", False),
+    )
+    def test_is_view(self, sql, expected):
+        is_(self.parser._check_view(sql), expected)

```

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 83db7a2780fd15e878930d07ecdb151da195577f of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 83db7a2780fd15e878930d07ecdb151da195577f: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4108

File: test/dialect/mysql/test_reflection.py [hunk 1]
```diff
@@ -1412,3 +1410,14 @@ def test_fk_reflection(self):
                 "SET NULL",
             ),
         )
+
+    @testing.combinations(
+        (
+            "CREATE ALGORITHM=UNDEFINED DEFINER=`scott`@`%` SQL SECURITY DEFINER VIEW `v1` AS SELECT",  # noqa: E501
```
Comment (zzzeek) [inline test/dialect/mysql/test_reflection.py:1416]: oh you can break this line up like this:

```py
(
    "CREATE ALGORITHM=UNDEFINED DEFINER=`scott`@`%` "
    "SQL SECURITY DEFINER VIEW `v1` AS SELECT",
)
```

we prefer to do it that way

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 31b98671b895eb166c108f7448d0f0985ba15c13 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 31b98671b895eb166c108f7448d0f0985ba15c13 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4108

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

OK things this needs:

1. split long line in half rather than doing a noqa
2. First line of the commit should be "fix MySQL "CREATE VIEW" regex" or somethign like that
3. the "fixes:" indicator needs the number sign:    `Fixes: #8588`
4. this will be a 1.4 backport, we can do that here
5. needs a changelog file doc/build/changelog/unreleased_14/8588.rst

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4108

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

note to submitter, feel free to hand it off to us for these things which I can do in under a minute, we'll still credit you for the PR, thanks for the contribution!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4108

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision d85b2c5b51e45cec543c9ae9d62d6d659b063354 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset d85b2c5b51e45cec543c9ae9d62d6d659b063354 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4108

**Comment by john-bodley** [comment]
@zzzeek thanks for the review. I've hopefully addressed your comments.

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4118 has been **merged**. Congratulations! :)

**Pull request closed without merging**